src/elf_sections.rs

```diff
@@ -62,6 +62,29 @@ pub struct ElfSection {
 }
 
 impl ElfSection {
+    pub fn name(&self, tag: &ElfSectionsTag) -> &'static str {
+        use core::{str, slice};
+
+        let strtabs = unsafe {
+            &*(&tag.first_section as *const ElfSection).offset(tag.shndx as isize)
+        };
+
+        let name_byte = (strtabs.addr + self.name as u64) as *const _;
+
+        unsafe {
+            let mut strlen = 0;
+            for i in 0.. {
+                if *name_byte.offset(i) == 0 {
+                    strlen = i as usize;
+                    break;
+                }
+            }
+
+            str::from_utf8_unchecked(
```

Comment:
Maybe we should use the safe [from_utf8](https://doc.rust-lang.org/core/str/fn.from_utf8.html) instead? I'm not sure if the specification guarantees that the string is valid utf8… And even if it does, I think that the additional check is a good idea, because performance shouldn't really matter in this case.